Sources/WalletConnectModal/Modal/ModalSheet.swift

```diff
@@ -81,14 +88,31 @@ public struct ModalSheet: View {
         .overlay(
             VStack {
                 if viewModel.destination.hasSearch {
-                    TextField("Search", text: $viewModel.searchTerm)
-                        .transform {
-                            #if os(iOS)
-                            $0.textFieldStyle(.roundedBorder)
-                                .autocapitalization(.none)
-                            #endif
-                        }
-                        .padding(.horizontal, 50)
+                    
+                    HStack {
+                        Image(systemName: "magnifyingglass")
+                        TextField("Search", text: $viewModel.searchTerm, onEditingChanged: { editing in
+                            self.searchEditing = editing
+                        })
+                    }
+                    .padding(.vertical, 4)
+                    .padding(.horizontal, 10)
+                    .background(Color.background3)
+                    .foregroundColor(searchEditing ? .foreground1 : .foreground3)
+                    .cornerRadius(28)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 28)
+                            .stroke(searchEditing ? Color.accent : Color.thinOverlay, lineWidth: 1)
+                    )
+                    .onDisappear {
```

Comment:
Can we move it on parent view from `body` ?

Reply:
It's related only to `TextField`s  onDisappear , otherwise I would agree with you to put these lifecycle modifier to top most view